core/templates/domain/platform_feature/client-context-object.factory.ts

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating ClientContext domain objects.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+export interface ClientContextBackendDict {
+  'client_type': string;
+  'browser_type': string;
+  'app_version'?: string;
```

Comment:
Typically we don't want these to be optional. Can it be made required? There is additional context that needs to be provided for optional strings, maybe talk to @nishantwrp if you need to do this.

Reply:
In fact I think we can just delete 'app_version' since it's never used in the frontend. -- deleted it in the latest commit.